factory-containers/assemble-system-image

```diff
@@ -146,48 +148,69 @@ def dump_apps_container_images(target: dict, containers_arch: str, app_image_tar
-        input=b'Yes\n')
-    p.check_returncode()
-
-    cmd('losetup', '-P', '-f', wic_image)
-    out = cmd('losetup', '-a', capture=True).decode()
-    for line in out.splitlines():
-        if wic_image in line:
-            loop = out.split(':', 1)[0]
-            break
-    else:
-        raise RuntimeError('Unable to find loop device for wic image')
+                                   '-C', image_data_dir[containers_arch], '.'])
+            return int(subprocess.check_output(['du', '-sk', image_data_dir[containers_arch]]).split()[0].decode('utf-8'))
+
+
+class WicImage:
+    def __init__(self, wic_image_path: str, increase_bytes=None, extra_space=0.2):
+        self._path = wic_image_path
+        self._mnt_dir = os.path.join('/mnt', 'wic_image_p2')
+        self._resized_image = False
+        if increase_bytes:
+            self._resize_wic_file(increase_bytes, extra_space)
+            self._resized_image = True
+
+    def __enter__(self):
+        cmd('losetup', '-P', '-f', self._path)
+        out = cmd('losetup', '-a', capture=True).decode()
+        for line in out.splitlines():
+            if self._path in line:
+                self._loop_device = out.split(':', 1)[0]
+                self._wic_device = out.split(':', 1)[0] + 'p2'
+                break
+        else:
+            raise RuntimeError('Unable to find loop device for wic image')
 
-    try:
         # containers don't see changes to /dev, so we have to hack around
         # this by basically mounting a new /dev. The idea was inspired by
         # this comment:
         #  https://github.com/moby/moby/issues/27886#issuecomment-257244027
         cmd('mount', '-t', 'devtmpfs', 'devtmpfs', '/dev')
-        cmd('e2fsck', '-y', '-f', loop + 'p2')
-        cmd('resize2fs', loop + 'p2')
-    finally:
-        cmd('losetup', '-d', loop)
+        cmd('e2fsck', '-y', '-f', self._wic_device)
+
+        if self._resized_image:
+            cmd('resize2fs', self._wic_device)
+
+        os.mkdir(self._mnt_dir)
+        cmd('mount', self._wic_device, self._mnt_dir)
+        return os.path.join(self._mnt_dir, 'ostree/deploy/lmp')
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        cmd('umount', self._mnt_dir)
+        os.rmdir(self._mnt_dir)
+        cmd('losetup', '-d', self._loop_device)
+
+    def _resize_wic_file(self, increase_bytes: int, extra_space=0.2):
+        bs = 1024
+        increase_k = ceil((increase_bytes + increase_bytes * extra_space) / bs) + 1
+        wic_k = ceil(os.stat(self._path).st_size / bs)
+        logger.info('Adding %d bytes to wic size (asked %d)', increase_k * bs, increase_bytes)
+        cmd('dd', 'if=/dev/zero', 'bs=' + str(bs), 'of=' + self._path,
+            'conv=notrunc', 'oflag=append', 'count=' + str(increase_k),
+            'seek=' + str(wic_k))
+
+        p = subprocess.run(
+            ['parted', '---pretend-input-tty', self._path, 'resizepart', '2', '100%'],
+            input=b'Yes\n')
```

Comment:
this is copied from me, but should this be `b'Fix\n'` instead of "yes"?

Reply:
`sgdisk -e` fixes the issue. It should go after this PR https://github.com/foundriesio/lmp-images-tools/pull/5 is merged and the image is built.